apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/service/PageHistoryFinder.java

```diff
@@ -0,0 +1,59 @@
+package gov.cdc.nbs.questionbank.page.service;
+
+import gov.cdc.nbs.questionbank.page.model.PageHistory;
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Service;
+
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class PageHistoryFinder {
+    private final JdbcTemplate jdbcTemplate;
+
+    private static final DateTimeFormatter dateFormatter = DateTimeFormatter
+            .ofPattern("MM/dd/yyyy")
+            .withZone(ZoneId.of("UTC"));
+
+    private static final String QUERY = """
+            SELECT
+                      hist.publish_version_nbr AS publishVersionNbr,
+                      hist.last_chg_time AS lastUpdatedDate,
+                      CONCAT(userProfile.first_nm, ' ', userProfile.last_nm) AS lastUpdatedBy,
+                      hist.version_note AS notes
+                    FROM
+                      WA_template_hist hist
+                      LEFT OUTER JOIN user_profile userProfile ON hist.last_chg_user_id = userProfile.nedss_entry_id
+                    WHERE
+                      hist.wa_template_uid = ?
+                    UNION
+                    SELECT
+                      tem.publish_version_nbr AS publishVersionNbr,
+                      tem.last_chg_time AS lastUpdatedDate,
+                      CONCAT(userProfile.first_nm, ' ', userProfile.last_nm) AS lastUpdatedBy,
+                      tem.version_note AS notes
+                    FROM
+                      WA_template tem
+                      LEFT OUTER JOIN user_profile userProfile ON tem.last_chg_user_id = userProfile.nedss_entry_id
+                    WHERE
+                      tem.wa_template_uid = ?
```

Comment:
issue:  When a page is published and a draft is created there are essentially two versions of the same Page, a published and a draft, each having their own identifier.  This means that `wa_template_uid` cannot be used to get all of the history.  The `template_nm` is the most unique common field.  

Something like...

```sql
with [target] ([name]) as (
    select 
        [target].template_nm
    from WA_template [target]
    where [target].[wa_template_uid] = ?
),
select
...
from [target]

    join wa_template_hist [history] on
        [target].[name] = [history].template_nm
...
UNION
select
...
from [target]

    join wa_template [page] on
        [page].template_nm = [target].[name]
```

suggestion:  The interesting part is that the API is passing only the identifier of the Page we want history for.  This can be handled in SQL using a CTE that grabs the `template_nm` of the Page using the `wa_template_uid` and then join the other two tables to it using the name.